temporian/core/operators/filter.py

```diff
@@ -93,3 +95,37 @@ def filter(
     assert isinstance(condition, EventSetNode)
 
     return FilterOperator(input, condition).outputs["output"]
+
+
+@compile
+def before(
+    input: EventSetOrNode, timestamp: Union[float, datetime]
+) -> EventSetOrNode:
+    assert isinstance(input, EventSetNode)
+
+    if isinstance(timestamp, datetime):
+        if not input.schema.is_unix_timestamp:
+            raise ValueError(
```

Comment:
Do we want the opposite to be allowed? e.g. timestamp is a float and the input is a unix timestamp.